jupyterlab_pullrequests/managers/manager.py

```diff
@@ -112,3 +116,75 @@ def read_notebook(content):
         )
 
         return {"base": prev_nb, "diff": thediff}
+
+    async def _call_service(
+        self,
+        url: str,
+        load_json: bool = True,
+        method: str = "GET",
+        body=None,
+        params: Optional[Dict[str, str]] = None,
+        headers: Optional[Dict[str, str]] = None,
+    ):
+        if not self._access_token:
+            raise tornado.web.HTTPError(
+                status_code=http.HTTPStatus.BAD_REQUEST,
+                reason="No access token specified.",
+            )
+
+        if body is not None:
+            if headers is None:
+                headers = {}
+            headers["Content-Type"] = "application/json"
+            body = json.dumps(body)
+
+        if not url.startswith(self._base_api_url):
+            url = "/".join((self._base_api_url.rstrip("/"), url.lstrip("/")))
+
+        if params is not None:
+            url = tornado.httputil.url_concat(url, params)
+
+        # User agents required for Github API, see https://developer.github.com/v3/#user-agent-required
+        try:
+            request = tornado.httpclient.HTTPRequest(
+                url,
+                validate_cert=True,
+                user_agent="JupyterLab Pull Requests",
+                method=method.upper(),
+                body=body,
+                headers=headers,
+            )
+        except BaseException as e:
```

Comment:
catching `BaseException` has a bad smell...

Reply:
I decided to remove that try...except as the creation of the request object should not fail as its input are controlled by design.